### ISSUE TITLE
Possible memory leaks in `point_new`

### ISSUE THREAD

**Issue by QthCN** [tree-sitter/py-tree-sitter#49]
Hi, I'm debugging the memory increase issue in our product(which use py-tree-sitter) and found there may exist a memory leak problem in `point_new`:

```
static PyObject *point_new(TSPoint point) {
  PyObject *row = PyLong_FromSize_t((size_t)point.row);
  PyObject *column = PyLong_FromSize_t((size_t)point.column);
  if (!row || !column) {
    Py_XDECREF(row);
    Py_XDECREF(column);
    return NULL;
  }
  return PyTuple_Pack(2, row, column);
}
```

`PyTuple_Pack` will add a reference on its arguments:

```
PyObject *
PyTuple_Pack(Py_ssize_t n, ...)
{
    Py_ssize_t i;
    PyObject *o;
    PyObject **items;
    va_list vargs;

    if (n == 0) {
        return tuple_get_empty();
    }

    va_start(vargs, n);
    PyTupleObject *result = tuple_alloc(n);
    if (result == NULL) {
        va_end(vargs);
        return NULL;
    }
    items = result->ob_item;
    for (i = 0; i < n; i++) {
        o = va_arg(vargs, PyObject *);
        Py_INCREF(o);
        items[i] = o;
    }
    va_end(vargs);
    tuple_gc_track(result);
    return (PyObject *)result;
}
```

So after I change point_new to following code, our product's memory usage is stable:

```
static PyObject *point_new(TSPoint point)
{
  PyObject *row = PyLong_FromSize_t((size_t)point.row);
  PyObject *column = PyLong_FromSize_t((size_t)point.column);
  if (!row || !column)
  {
    Py_XDECREF(row);
    Py_XDECREF(column);
    return NULL;
  }

  PyObject *obj = PyTuple_Pack(2, row, column);
  Py_XDECREF(row);
  Py_XDECREF(column);
  return obj;
}
```

I'm not very familiar with the python c extension, so not very sure if it's a real memory leak.

**Comment by maxbrunsfeld** [comment]
⚡  Thanks for the great debugging and explanation.

Are you interested in opening a pull request that implements your fix?

**Comment by QthCN** [comment]
> ⚡ Thanks for the great debugging and explanation.
> 
> Are you interested in opening a pull request that implements your fix?

Sure. I open a PR #50 with the fix.

**Comment by maxbrunsfeld** [comment]
This is included in tree_sitter 0.2.2.
